File: ocr_script_hoang_tu_ech.py
import os
import requests
import json
import time

API_KEY = os.environ.get("FILEGRAPH_API_KEY")
API_BASE = "https://api.filegraph.ai"

def ocr_image(image_path):
    url = f"{API_BASE}/image/to-text"
    headers = {"X-API-Key": API_KEY}
    with open(image_path, "rb") as f:
        files = {"file": f}
        # Theo rule: tiếng Việt (vie)
        params = {"language": "vie"}

        response = requests.post(url, headers=headers, files=files, params=params)
    if response.status_code == 200:
        return response.json().get("text", "")
    else:
        print(f"Error OCR {image_path}: {response.text}")
        return ""

def main():
    results = {}
    images = sorted([f for f in os.listdir("temp_hoang_tu_ech") if f.endswith(".jpg")],
                    key=lambda x: int(x.split("-")[-1].split(".")[0]))

    for img in images:
        path = os.path.join("temp_hoang_tu_ech", img)
        print(f"Processing {path}...")
        text = ocr_image(path)
        results[img] = text
        time.sleep(1) # Rate limit safety

    with open("ocr_results_hoang_tu_ech.json", "w", encoding="utf-8") as f:
        json.dump(results, f, ensure_ascii=False, indent=2)

if __name__ == "__main__":
    main()

File: ocr_script_hoang_tu_ech.js
import fs from 'fs';
import path from 'path';
import axios from 'axios';
import 'dotenv/config';

const API_KEY = 'fg_live_PBMLwShJTRR8AKdfAazH4P0iIKUDU7rg';
const API_BASE = 'https://api.filegraph.ai';

async function ocrImage(imagePath) {
    const url = `${API_BASE}/image/to-text`;
    const fileBuffer = fs.readFileSync(imagePath);
    const blob = new Blob([fileBuffer], { type: 'image/jpeg' });
    const formData = new FormData();
    formData.append('file', blob, path.basename(imagePath));

    try {
        const response = await axios.post(url, formData, {
            headers: {
                'X-API-Key': API_KEY,
            },
            params: {
                language: 'eng'
            }
        });
        return response.data.text || '';
    } catch (error) {
        console.error(`Error OCR ${imagePath}:`, error.response?.data || error.message);
        return '';
    }
}

async function main() {
    const dir = 'temp_hoang_tu_ech';
    const results = {};
    const files = fs.readdirSync(dir)
        .filter(f => f.endsWith('.jpg'))
        .sort((a, b) => {
            const numA = parseInt(a.match(/(\d+)/)[0]);
            const numB = parseInt(b.match(/(\d+)/)[0]);
            return numA - numB;
        });

    for (const file of files) {
        const filePath = path.join(dir, file);
        console.log(`Processing ${filePath}...`);
        const text = await ocrImage(filePath);
        results[file] = text;
        await new Promise(r => setTimeout(r, 1000));
    }

    fs.writeFileSync('ocr_results_hoang_tu_ech.json', JSON.stringify(results, null, 2), 'utf-8');
}

main().catch(console.error);
